packages/header-component/src/components/dc-header/dc-header.tsx

```diff
@@ -94,6 +99,16 @@ export class Header {
     this._links = JSON.parse(newValue);
   }
 
+  @Watch("user")
+  userDidChage(newUser) {
```

Comment:
As two objects are going to be different `{} === {} // false` wouldn't this lead to the same issue we used to have?.
While may still have some edge cases, first thing that comes to my mind would be to compare the users' id.

**On top of that,** it feels awkward to mutate the prop like that, **what about something on the render instead?** It is not as sexy, but the main goal is not to mutate the prop value.

Perhaps on https://github.com/debtcollective/packages/blob/b520e1e1d610900ad2d52607bb5cba9773cd7318/packages/header-component/src/components/dc-header/dc-header.tsx#L190 you can do like `{ user.id ? <authLinks> : null }`